### ISSUE TITLE
Connection aborted, BadStatusLine

### ISSUE THREAD

**Issue by marekbedy** [danb35/deploy-freenas#3]
Hello,

your script crashes here:

root@JAIL-01:/certificates # ./deploy_freenas.py 
Certificate import successful
Certificate list successful
Setting active certificate successful
Traceback (most recent call last):
  File "./deploy_freenas.py", line 101, in <module>
    auth=(USER, PASSWORD),
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 490, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))

**Comment by danb35** [comment]
I believe this is normal, due to the way the API call to restart the web server works.  After the script finishes, is the web GUI using the new certificate?

**Comment by marekbedy** [comment]
The script has imported only server's certificate, the letsencrypt CA is not imported. Is this expected?

**Comment by danb35** [comment]
Do you mean the CA isn't imported as part of the certificate, or that it isn't imported into the Certificate Authorities section in the web GUI?  If the latter, that's expected; that just isn't part of what this script is trying to do.  If the former, sounds like a bug in FreeNAS--it had been reported some time ago, but I thought it was fixed a while back.

**Comment by danb35** [comment]
OK, as to importing the CAs, the script is working as designed--it just isn't designed to do that.  It might be worthwhile to figure out how, but it'll take some doing.

Failing connections isn't a good thing, though.  From the shell in FreeNAS, run `openssl s_client -connect localhost:443`.  What's the output?  Mask your domain name if you like.

**Comment by marekbedy** [comment]
root@ucw:~ # openssl s_client -connect localhost:443
connect: Connection refused
connect:errno=61
root@ucw:~ #

**Comment by marekbedy** [comment]
root@ucw:~ # sockstat -4 | grep nginx
www      nginx      2789  6  tcp4   195.20.170.17:80      *:*
www      nginx      2789  14 tcp4   195.20.170.17:80      195.20.170.5:63227
www      nginx      2789  17 tcp4   127.0.0.1:18940       127.0.0.1:6000
root     nginx      2788  6  tcp4   195.20.170.17:80      *:*

**Comment by danb35** [comment]
And you've set the web GUI to use HTTPS (HTTP+HTTPS would be safer, at least while you're trying to get it to work)?
![image](https://user-images.githubusercontent.com/12121759/38627020-e8b29676-3d7b-11e8-9518-b0d1b4fa44cf.png)

**Comment by marekbedy** [comment]
yes, http+ https enabled. It seems nxginx daemon (GUI web server) does not start https

![image](https://user-images.githubusercontent.com/5310638/38627151-954ae014-3dae-11e8-84b8-35b428d99b63.png)

**Comment by danb35** [comment]
It shouldn't be required, but can you try restarting the server?

**Comment by marekbedy** [comment]
When i restart nginx service manualy i got this:
root@ucw:~ # service nginx restart
Performing sanity check on nginx configuration:
nginx: the configuration file /usr/local/etc/nginx/nginx.conf syntax is ok
nginx: configuration file /usr/local/etc/nginx/nginx.conf test is successful

But in the configuration it is not configured port 443:

more /usr/local/etc/nginx/nginx.conf
--- snip
    server {
        server_name  localhost;
        listen       0.0.0.0:80;
        listen       [::]:80;
--- snip

**Comment by marekbedy** [comment]
The restart of the server brings the same result.

**Comment by psmith1303** [comment]
... I've yet to master this github thing ... anyhow my fix to tidy this error  up is:

add this by the imports

``` python
#
# Some libs just for error handling
#
import urllib3, http

``` 


And then at the end where the reset of http is done, change it to:

``` python
# Reload nginx with new cert

try:
  r = requests.post(
    PROTOCOL + DOMAIN_NAME + '/api/v1.0/system/settings/restart-httpd-all/',
    auth=(USER, PASSWORD),
  )
except (urllib3.exceptions.ProtocolError, http.client.RemoteDisconnected, requests.exceptions.ConnectionError):
  print("Server reset ... as expected")
```

**Comment by fritz-fritz** [comment]
@danb35 can't this be closed by PR #6?
